CMakeLists.txt

```diff
@@ -17,6 +17,10 @@ if(NOT Poco_FOUND)
     list(APPEND extra_cmake_args "-DCMAKE_CXX_FLAGS=/wd4244 /wd4530 /wd4577")
   else()
     list(APPEND extra_cmake_args "-DCMAKE_C_FLAGS=-Wno-shift-negative-value")
+    if(APPLE)
+      # Starting with macOS 10.12, the use of OSAtomic*32 funcs are deprecated.
+      list(APPEND extra_cmake_args "-DCMAKE_CXX_FLAGS=-Wno-deprecated")
```

Comment:
No, apparently that is a macOS specific warning that was added to the compiler as a patch and is only affected by this higher level one. It's not exactly what they're talking about here, but it is the same issue:

http://stackoverflow.com/a/13461291/671658

Also, this is only for `APPLE`, so it shouldn't suppress things on Windows. I'm going to open an issue/pull request to poco when I get time, since this is a simple fix.